Blizzard.IO.Core/ITypeDeserializer.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Blizzard.IO.Core
+{
+    public interface ITypeDeserializer<TData>
```

Comment:
Myabe Call it ConcreteTypeDeserializer?
```suggestion
    public interface IConcreteTypeDeserializer<TData>
```